### PR TITLE
Updated `wp-env` To 8.0.0

### PR DESCRIPTION
**Pull request by ObliviousHarmony**
### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

We are updating `@wordpress/env` to 8.0.0 in order to consume some upstream fixes.

- `wp-env` no longer requires WSL to function.
- Fixes to `wp-env run` that no longer require us to use quotes (and in fact, break, when we do)

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Run `pnpm --filter=woocommerce env:test` after updating.
2. Run `pnpm --filter=woocommerce test:unit:env` and confirm they pass.
3. Run `pnpm --filter=woocommerce test:e2e-pw` and confirm they pass.

### CODE REVIEW

**Comment by ObliviousHarmony** [comment]
The `code-analyzer` is broken in this PR because of a breaking change in `wp-env`. Once it gets merged subsequent PRs won't have this problem.

**Review by lsinger — COMMENTED** [review]
The changes look good to me, CI passes (with one expected and acceptable exception), but locally I always have one or two e2e tests fail even after retries. Not sure if that's acceptable / expected / OK so I'd like to hear the perspective from Solaris.

File: plugins/woocommerce/.wp-env.json [hunk 1]
```diff
@@ -12,7 +12,10 @@
     "mappings": {
         "wp-cli.yml": "./tests/wp-cli.yml"
     },
-    "afterSetup": "./tests/e2e-pw/bin/test-env-setup.sh",
+    "lifecycleScripts": {
+        "afterStart": "./tests/e2e-pw/bin/test-env-setup.sh",
+        "afterClean": "./tests/e2e-pw/bin/test-env-setup.sh"
```
Comment (rodelgc) [inline plugins/woocommerce/.wp-env.json:17]: Thank you for adding the `afterClean` scenario here!

**Comment by rodelgc** [comment]
Looking good! I ran the tests and GitHub workflows that are likely to be affected by this change.

- :white_check_mark: Tested the `upload-plugin.spec.js` E2E test locally
- :white_check_mark: Tested the `update-woocommerce.spec.js` E2E test locally
- :white_check_mark: Launched wp-env environment using `env:test:cot` and ran `basic.spec.js`
- :white_check_mark: [Ran](https://github.com/woocommerce/woocommerce/actions/runs/5090926619/jobs/9150343611) _Smoke test release_ workflow on a clone of this branch. The "Test WooCommerce update" and "With (Plugin name)" jobs passed.
- :white_check_mark: [Ran](https://github.com/woocommerce/woocommerce/actions/runs/5090910723/jobs/9150520858) _Smoke test daily_ workflow on a clone of this branch. The jobs "E2E on nightly build" and "Smoke test with (plugin name) installed" passed.

Seems like there are conflicts in the PNPM lock file. I'm pre-emptively approving this PR in case the conflict isn't related to the changes here.

**Review by rodelgc — DISMISSED** [review]

**Review by jonathansadowski — APPROVED** [review]
👍 looks good to me with the noted caveat about the code-analyzer failure being related to this change, which seems low-risk as this PR doesn't change any templates 😄

**Pull request merged**